drivers/flash/nrf_qspi_nor.c

```diff
@@ -507,6 +529,12 @@ static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 	int ret = nrfx_qspi_read(dest, size, addr);
 
 	qspi_wait_for_completion(dev);
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
```

Comment:
We are missing calls to `k_free` on some returns. Please consider using the `goto out` pattern.

Reply:
Added